internal/ethapi/api.go

```diff
@@ -1135,23 +1135,9 @@ func (s *PublicTransactionPoolAPI) SendRawTransaction(ctx context.Context, encod
 		return "", err
 	}
 
-	if err := s.b.SendTx(ctx, tx); err != nil {
-		return "", err
-	}
-
-	signer := types.MakeSigner(s.b.ChainConfig(), s.b.CurrentBlock().Number())
-	if tx.To() == nil {
-		from, err := types.Sender(signer, tx)
-		if err != nil {
-			return "", err
```

Comment:
This check is not present in `submitTransaction`. Maybe it should be, but when `submitTransaction` is called internally, we've just signed it ourselves (so a 'soft guarantee' that it will be a valid signature). When it's called from `sendRawTransaction`, there's no such soft guarantee.

Reply:
yes, you are right! I fixed it again, maybe now it is ok.